baybe/searchspace/continuous.py

```diff
@@ -91,7 +95,7 @@ def from_bounds(cls, bounds: pd.DataFrame) -> SubspaceContinuous:
 
         # Create the corresponding parameters and from them the search space
         parameters = [
-            NumericalContinuousParameter(name, bound)
+            NumericalContinuousParameter(cast(str, name), bound)
```

Comment:
Why is this explicit cast here necessary?

Reply:
That's because name comes from the columns of a pandas DataFrame and they can contain arbitrary "Hashables", so mypy correctly complains that the input is not necessarily a string. But since we know it will always ever be a string (since we populate it with strings), the `cast` takes care of it. As a reminder: `cast` is only for static typing, not evaluated during runtime. The alternative would be to use an actual `str` conversion, but I thought the above is more appropriate